charts/redpanda/templates/tests/test-connector-via-console.yaml

```diff
@@ -122,61 +128,37 @@ spec:
           EOF
 
           sed -i "s/CONNECTOR_NAME/$CONNECTOR_NAME/g" /tmp/mm2-conf.json
-          sed -i "s/SASL_MECHANISM/$SASL_MECHANISM/g" /tmp/mm2-conf.json
+          sed -i "s/SASL_MECHANISM/$RPK_SASL_MECHANISM/g" /tmp/mm2-conf.json
           sed -i "s/SECURITY_PROTOCOL/$SECURITY_PROTOCOL/g" /tmp/mm2-conf.json
           set +x
           sed -i "s/JAAS_CONFIG_SOURCE/$JAAS_CONFIG_SOURCE/g" /tmp/mm2-conf.json
           sed -i "s/JAAS_CONFIG_TARGET/$JAAS_CONFIG_TARGET/g" /tmp/mm2-conf.json
           set -x
 
-          max_iteration=10
-          for i in $(seq 1 $max_iteration)
-          do
-            curl -v -H 'Content-Type: application/json' http://{{ include "console.fullname" $consoleValues }}:{{ include "console.containerPort" $consoleValues }}/api/kafka-connect/clusters/connectors/connectors \
-            -d @/tmp/mm2-conf.json && echo
-
-            result=$?
-            if [[ $result -eq 0 ]]
-            then
-              echo "Result successful"
-              break
-            else
-              echo "Result unsuccessful"
-              sleep 1
-            fi
-          done
-
-          if [[ $result -ne 0 ]]
+          URL=http://{{ include "console.fullname" $consoleValues }}:{{ include "console.containerPort" $consoleValues }}/api/kafka-connect/clusters/connectors/connectors
+          {{/* outputting to /dev/null because the output contains the user password */}}
+          echo "Creating mm2 connector"
+          if curl -svm3 --fail --retry 120 --retry-max-time 120 --retry-all-errors -H 'Content-Type: application/json' -o /dev/null "${URL}" -d @/tmp/mm2-conf.json
```

Comment:
Please don't use this short hand as it is incredible hard to debug failures. Curl will eat and not inform how each request failed with what error and what HTTP code.

Reply:
is it though? doesnt the verbosity aid in trouble shooting? I guess do we have an example on the difficulty for debugging?

Reply:
It's verbose, so all the retries will be printed. Besides, if it fails until it succeeds, we don't care. If it fails until it fails, that last failure is all we need to know anyway. That'll tell us why it failed.

Reply:
> I guess do we have an example on the difficulty for debugging?

https://github.com/redpanda-data/helm-charts/pull/567/commits/cea5b4ddf09076912827a0092abb8601a32a7bd3
https://github.com/redpanda-data/helm-charts/pull/567

based on https://redpandadata.slack.com/archives/C04K2N6S2A3/p1687526434321209

> doesnt the verbosity aid in trouble shooting?

If you would like to trace what happen and what HTTP codes where returned you need verbosity.

> It's verbose

No, as curl has `-svm3`. The `v` stands to verbose, but `s` is for silent which will take precedence. 

```
Silent or quiet mode. Do not show progress meter or error messages. Makes Curl mute. It still outputs the data you ask for, potentially even to the terminal/stdout unless you redirect it.

Use [-S, --show-error](https://curl.se/docs/manpage.html#-S) in addition to this option to disable progress meter but still show error messages.
```
https://curl.se/docs/manpage.html#-s

If we would use upper case S maybe all error would be returned.

When we will hit the error I will come back to this message as for me problems in CI are not easily reproducible.